pkg/internal/apis/certmanager/validation/issuer.go

```diff
@@ -264,9 +264,8 @@ func ValidateACMEIssuerDNS01Config(iss *v1alpha1.ACMEIssuerDNS01Config, fldPath
 				el = append(el, field.Forbidden(fldPath.Child("azuredns"), "may not specify more than one provider type"))
 			} else {
 				numProviders++
-				el = append(el, ValidateSecretKeySelector(&p.AzureDNS.ClientSecret, fldPath.Child("azuredns", "clientSecretSecretRef"))...)
-				if len(p.AzureDNS.ClientID) == 0 {
-					el = append(el, field.Required(fldPath.Child("azuredns", "clientID"), ""))
+				if len(p.AzureDNS.ClientID) > 0 {
+					el = append(el, ValidateSecretKeySelector(&p.AzureDNS.ClientSecret, fldPath.Child("azuredns", "clientSecretSecretRef"))...)
```

Comment:
I think we should amend this check to test:

* if the client ID is set, the clientSecret is required and must be valid
* if the clientSecret is set, clientID must be provided and must be valid
* if neither are set, nothing should be checked (i.e. ClientID is no longer required)